rtos/mbed_boot.c

```diff
@@ -413,6 +415,45 @@ void __rt_entry (void) {
     mbed_start_main();
 }
 
+typedef void *mutex;
+
+/* ARMC requires variable number of dynamically created mutexes, unfortunately mbed OS
+   doesn't support RTX dynamic object creation. To work around this limitation we enable
+   RTX memory pool, with maximum size set to 5, which is necessary for the platform to
+   boot. When toolchain exhausts the reserved pool, we are switching to allocating
+   storage memory on the heap. We can't use the heap approach exclusively because
+   ARMC protects malloc calls with a mutex. We don't need to worry about freeing the
+   allocated memory as library mutexes seem never to be actually deleted.
+ */
+int _mutex_initialize(mutex *m)
+{
+    osMutexAttr_t attr;
+    memset(&attr, 0, sizeof(attr));
+    attr.name = "ARMC toolchain mutex";
+    attr.attr_bits = osMutexRecursive | osMutexPrioInherit | osMutexRobust;
+
+    *m = osMutexNew(&attr);
```

Comment:
if osMutexNew succeeds, then you have 2 calls to osMutexNew(&attr)
one here and another one below. 
so you have *m overridden with another allocation and previous one leaks
is that intentional?